services/app-web/src/components/ChangeHistory/ChangeHistory.tsx

```diff
@@ -93,27 +119,21 @@ export const ChangeHistory = ({
     }
 
     const revisionHistory = flattenedRevisions()
-
     const revisedItems: AccordionItemProps[] = revisionHistory.map(
         (r, index) => {
             const isInitialSubmission = r.updatedReason === 'Initial submission'
-            const isSubsequentSubmission = r.kind === 'submit'
-
+            const isSubsequentSubmissionOrUnlock =
+                r.kind === 'submit' || r.kind === 'unlock'
+            const isApprovalAction = r.kind === 'approve'
             // We want to know if this contract has multiple submissions. To have multiple submissions, there must be minimum
             // more than the initial contract revision.
             const hasSubsequentSubmissions = revisionHistory.length > 1
 
-            return {
-                title: (
-                    <div>
-                        {`${formatToPacificTime(r.updatedAt)} - ${isSubsequentSubmission ? 'Submission' : 'Unlock'}`}
-                    </div>
-                ),
-                // Display this code if this is the initial contract. We only want to display the link of the initial contract
-                // only if there has been subsequent contracts. We do not want to display a link if the package initial
-                // contract was unlocked, but has not been resubmitted yet.
-                headingLevel: 'h4',
-                content: isInitialSubmission ? (
+            let content
+            let title
+            if (isInitialSubmission) {
```

Comment:
This has enough choices in it it might be cleaner to break out as a separate function that returns content and title.